urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/util/FileLogHandler.java

```diff
@@ -0,0 +1,183 @@
+     *
+     * @param pattern 
+     * @param append
+     * @param limit 
+     **/
+    public FileLogHandler( String pattern,
+			   boolean append,
+			   int limit 
+			   ) 
+	throws NullPointerException,
+	       IllegalArgumentException {
+	this( pattern, 
+	      append, 
+	      limit, 
+	      new SimpleDateFormat(FileLogHandler.DEFAULT_DATE_FORMAT_PATTERN) 
+	      );
+    }
+
+    /**
+     * The constructor.
+     *
+     * @param pattern 
+     * @param append
+     * @param limit 
+     **/
+    public FileLogHandler( String pattern,
+			   boolean append,
+			   int limit,			   
+			   DateFormat dateFormat
+			   ) 
+	throws NullPointerException,
+	       IllegalArgumentException {
+
+	super();
+
+	if( pattern == null )
+	    throw new NullPointerException( "Cannot create FileLogHandlers with null pattern." );
+	if( dateFormat == null )
+	    throw new NullPointerException( "Cannot create FileLogHandlers with null date-format." );
+	if( limit <= 0 )
+	    throw new IllegalArgumentException( "Cannot create FileLoghandler with negative/zero max file limit." );
+	
+	this.pattern     = pattern;
+	this.append      = append;
+	this.limit       = limit;
+	
+	this.dateFormat  = dateFormat;
+
+	// Note there is the 'setOutputStream()' method!
+    }
+
+    //--- BEGIN --------------- Handler implementation ------------------------
+    public void close() 
+	throws SecurityException {
+	
+	synchronized( this.pattern ) {	    
+	    super.close();
+	}
+    }
+
+    
+    public void	flush() 
+	throws SecurityException {
+
+	synchronized( this.pattern ) {
+	    super.flush();	    
+	}
+    }
+
+    public void	publish( LogRecord record ) {
+	
+	synchronized( this.pattern ) {
+	    super.publish( record );
+	}
+    }
+    //--- END ----------------- Handler implementation ------------------------
+    
+
+    protected File getNextFileInCycle() {
+	
+	return null;
+    }
+
+    public static String getNextFilename( String pattern,
+					  DateFormat dateFormat,
+					  int uniqueNumber,
+					  int groupNumber ) {
+	
+	String filename = pattern;
+	filename = filename.replaceAll( "/", System.getProperty("file.separator") );
+	filename = filename.replaceAll( "%t", System.getProperty("java.io.tmpdir") );
+	filename = filename.replaceAll( "%u", Integer.toString(uniqueNumber) );
+	filename = filename.replaceAll( "%g", Integer.toString(groupNumber) );
+	filename = filename.replaceAll( "%h", System.getProperty("user.dir") );
+
+	// This is new
+	filename = filename.replaceAll( "%d", dateFormat.format(new Date(System.currentTimeMillis())) );
+
+	// Finally restore all '%'
+	filename = filename.replaceAll( "%%", "%" );
+	
+	return filename;
+    }
+
+
+    public static void main( String[] argv ) {
+
+	System.out.println( "Test run ..." );
+	String pattern = "%h/test/testlog.%d_u%u_g%g.log";
+	System.out.println( "Using pattern: " + pattern );
+	DateFormat dateFormat = new SimpleDateFormat(FileLogHandler.DEFAULT_DATE_FORMAT_PATTERN);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.